tutorials/nservicebus-101/lesson-5/tutorial.md

```diff
@@ -0,0 +1,195 @@
+System.Exception: BOOM!
+   at <stack trace>
+```
+
+10 seconds later, the retries begin again, followed by another yellow trace, sending the message back to delayed retries. 20 seconds after that, another set of traces. Finally, 30 seconds after that, the final exception trace will be shown in red, which is log level ERROR. This is where NServiceBus gives up on the message and redirects it to the error queue.
+
+```no-highlight
+INFO  Sales.PlaceOrderHandler Received PlaceOrder, OrderId = e927667c-b949-47ee-8ea2-f29523909784
+ERROR NServiceBus.RecoverabilityExecutor Moving message '53ac6836-48ef-49dd-aabb-a67c0104a2a5' to the error queue 'error' because processing failed due to an exception:
+System.Exception: BOOM!
+   at < stack trace>
+```
+
+
+### Modify immediate retries
+
+The only configurable option you need for immediate retries is how many of them to attempt. The default value is `5`, but you may want to set it to a higher or lower number. Many developers prefer to set it to `0` so that they can avoid the "wall of text" effect when an exception is thrown during development, and then set it to a higher number for production use.
+
+ 1. In the **Sales** endpoint, locate the **Program.cs** file.
+ 1. Before the endpoint is started, add the following code:
+
+snippet:ImmediateRetries
+
+Now, re-run the solution.
+
+Notice how delayed retries still occur (with exception traces in yellow) but no immediate retries (in white) occur between each delayed retry.
+
+Of course, the number of retries supplied to the immediate retries API can be pulled from an appSetting to allow changing configuration between development/test/staging/production environments.
+
+
+### Modify delayed retries
+
+Now let's try modifying the delayed retries. First let's try disabling them completely, which can also be beneficial during development to remove noise.
+
+Modify the recoverability code you wrote previously, then re-run the solution:
+
+snippet:DelayedRetries
+
+Notice how the message is only attempted once. The stack trace is red, and the message is immediately forwarded to the error queue.
+
+With delayed retries, we can also modify the time increase applied to each round of retries. Let's reset the number of delayed retries back to `3`, and decrease the time increase to 3 seconds so the delayed retries will occur very quickly.
+
+Modify the same recoverability code, and re-run the solution:
+
+snippet:TimeIncrease
+
+Notice how much faster the message proceeds through delayed retries, because instead of delays of 10/20/30 seconds (60 seconds total) the delays are now 3/6/9 seconds, for a total of 18 seconds wait time.
+
+
+### Replay a message
+
+If you used the [Particular Platform Installer](/platform/installer/) to install MSMQ, you should already have the [ServiceControl](/servicecontrol/) and [ServicePulse](/servicepulse/) tools installed.
+
+To check for these tools:
```

Comment:
the below steps are very fragile to changes in SC and SP. instead link to existing doco so it is managed in only one place